File: android_front/lib/mapper/hisotry/game_move_history_mapper.dart
import 'dart:convert';

import '../../model/history/game_move_history_model.dart';
import '../game/full_game_state_mapper.dart';

class GameMoveHistoryResponseMapper {

  static GameMoveHistory fromJson(Map<String, dynamic> json) {
    print(json['game_state']);
    Map<String, dynamic> game_state = jsonDecode(json['game_state']);
    return GameMoveHistory(
      id: json['id'],
      player: json['player'],
      game_state: FullGameStateMapper.fromJson(game_state),
      action: json['action'],
      action_number: json['action_number']
    );
  }
}

File: game-manager/src/database/mapper/game_move_history_mapper.rs
use crate::database::entity::game_move_history::GameMoveHistoryEntity;
use crate::domain::model::game_move_history::GameMoveHistory;

pub struct GameMoveHistoryEntityMapper;

impl GameMoveHistoryEntityMapper {

    pub fn entity_to_domain(game_move_history_entity: GameMoveHistoryEntity) -> GameMoveHistory {
        GameMoveHistory {
            id: game_move_history_entity.id,
            player: game_move_history_entity.player,
            game_state: game_move_history_entity.game_state,
            action: game_move_history_entity.action,
            action_number: game_move_history_entity.action_number,
            game_history_id: game_move_history_entity.game_history_id,
        }
    }
}